CMakeLists.txt

```diff
@@ -1,231 +1,264 @@
-    			"failed!\n\nSee \"FailedExecution.txt\" for more info."
-    		)
-    		file(WRITE ${CMAKE_BINARY_DIR}/FailedEasyRun.txt "${COMPILE_OUTPUT}\n${RUN_OUTPUT}")
-    	else()
-    		file(REMOVE ${CMAKE_BINARY_DIR}/FailedEasyRun.txt ${CMAKE_BINARY_DIR}/simple_xy.nc)
-    		set(HEMCO_TRY_RUN_PASSED TRUE CACHE INTERNAL "try_run passed" FORCE)
-    	endif()
+    # Display a warning if its execution failed
+    if(RUN_FAILED)
+        if(OMP)
+            set(CONDITIONAL_AND_OMP "and OpenMP ")
+        endif()
+        message(WARNING
+            "A simple program that uses NetCDF-Fortran "
+            "${CONDITIONAL_AND_OMP}compiled successfully, "
+            "but its execution failed!\n\nSee "
+            "\"FailedExecution.txt\" for more info."
+        )
+        file(WRITE ${CMAKE_BINARY_DIR}/FailedEasyRun.txt
+            "${COMPILE_OUTPUT}\n${RUN_OUTPUT}"
+        )
+    else()
+        file(REMOVE ${CMAKE_BINARY_DIR}/FailedEasyRun.txt
+            ${CMAKE_BINARY_DIR}/simple_xy.nc
+        )
+        set(GC_TRY_RUN_PASSED TRUE CACHE INTERNAL
+            "try_run passed" FORCE
+        )
     endif()
 endif()
 
-# Add all the subdirectories, each with its own CMakeLists.txt
 add_subdirectory(src)
 
-#############################################################
-# If using in GEOSgcm, uncomment lines below
-# NOTE: need to figure out a way to bypass that step,
-# in GEOSgcm and also in GCHP and GEOS-Chem Classic,
-# and any other model that uses HEMCO with cmake.
-#############################################################
-#esma_set_this ()
-#
-#set (src_directories
-#   ${CMAKE_CURRENT_SOURCE_DIR}
-#   ${CMAKE_CURRENT_SOURCE_DIR}/src/Shared
-#   ${CMAKE_CURRENT_SOURCE_DIR}/src/Core
-#   ${CMAKE_CURRENT_SOURCE_DIR}/src/Extensions
-#   ${CMAKE_CURRENT_SOURCE_DIR}/src/Interfaces
-#   )
-#
-#set (srcs)
-#foreach (dir ${src_directories})
-#  file (GLOB tmpsrcs ${dir}/*.[fF] ${dir}/*.[fF]90 ${dir}/*.c)
-#  list (APPEND srcs ${tmpsrcs})
-#endforeach()
-#
-## These files contain Fortran MODULE statements that are
-## deactivated by CPP statements.  CMake incorrectly determines
-## that the file produces the corresponding .mod file, but they do not.
-## Essentially a clash between using CPP to deactivate a file vs
-## the config system.  The latter is preferable.  TLC
-#list (REMOVE_ITEM srcs
-#   ${CMAKE_CURRENT_SOURCE_DIR}/src/Extensions/hcox_tomas_dustdead_mod.F
-#   ${CMAKE_CURRENT_SOURCE_DIR}/src/Extensions/hcox_tomas_seasalt_mod.F90
-#   ${CMAKE_CURRENT_SOURCE_DIR}/src/Extensions/hcox_tomas_jeagle_mod.F90
-#   )
-#
-#
-#esma_add_library (${this}
-#   SRCS ${srcs}
-#   DEPENDENCIES MAPL_Base
-#   )
-#
-#target_include_directories (${this} PUBLIC ${INC_ESMF} ${INC_NETCDF})
-#
-#target_compile_definitions (${this} PRIVATE ESMF_ DEVEL NO_BPCH UCX
-#   EXTERNAL_GRID EXTERNAL_FORCING GEOS_FP USE_REAL8 USE_TEND NC_DIAG DISCOVER MODEL_GEOS)
-#set_target_properties (${this} PROPERTIES Fortran_MODULE_DIRECTORY ${include_${this}})
+
```

Comment:
I don't think the try_compile code is necessary in HEMCO. That code was really just UX code for GEOS-Chem Classic to error during the CMake step if the user's NetCDF installation was broken. I think for HEMCO it's unnecessary complexity.

Reply:
This remove the need for CMakeScripts/try_compile.F90